### ISSUE TITLE
Build failed: Unrecognized VM option 'UseCGroupMemoryLimitForHeap'

### ISSUE THREAD

**Issue by duailibe** [react-native-community/react-native-circleci-orb#120]
### What happened

The build failed with the following error (in "Downloading Gradle Dependencies"):

```
Picked up _JAVA_OPTIONS: -Xmx2g -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap
Unrecognized VM option 'UseCGroupMemoryLimitForHeap'
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.

Exited with code exit status 1
```

I noticed this orb uses a non-pinned image of react-native-community/react-native, and that image [was updated](https://github.com/react-native-community/docker-android/commit/3c5a834aa4d6b5dd23a708e65011ad12e899e723) last week to use JDK 11 which completely removes the `UseCGroupMemoryLimitForHeap` option.

I'm not that familiar with JVM options so I just changed config.yml to pass `java_options` without that flag and the build succeeded. Moving forward I found [this article](https://merikan.com/2019/04/jvm-in-a-container/#java-10) which might help decide with the defaults should be.

**Comment by cortinico** [comment]
> I noticed this orb uses a non-pinned image of react-native-community/react-native

I believe that's an issue and should be fixed.

> and that image was updated last week to use JDK 11 which completely removes the UseCGroupMemoryLimitForHeap option.

True. If running on JDK11 the `UseCGroupMemoryLimitForHeap` option should be removed.
I think this orb should be update to:
- Release a version (say 5.7.0) which relies on a pinned version of the docker container that runs on JDK8 (anything on 4.x).
- Release another version (say 6.0.0) which relies on a pinned version of the docker container that runs on JDK11 (anything on 5.x).

**Comment by mikehardy** [comment]
In the meantime, we're dead in the water because of this line :-)

https://github.com/react-native-community/react-native-circleci-orb/blob/1b12f2a14a3bc7a5dba2ea7f340508b9d6b4cb2f/src/executors/linux_android.yml?#L5

I liked your suggestion @cortinico and I lined up two PRs #121 and #122 to implement it. Until they are done our CI is broken and we will have trouble validating releases, for example: 
https://github.com/react-native-share/react-native-share/pull/1124/checks?check_run_id=4251523347

**Comment by vonovak** [comment]
closed via #122. Please open a new issue if the problem persists :)

**Comment by mikehardy** [comment]
v6.0.1 of this orb is working correctly - comment with links to proof https://github.com/react-native-community/react-native-circleci-orb/pull/123#issuecomment-974095135
